base/nullable.jl

```diff
@@ -63,16 +63,79 @@ get(x::Nullable) = x.isnull ? throw(NullException()) : x.value
 
 isnull(x::Nullable) = x.isnull
 
-function isequal(x::Nullable, y::Nullable)
-    if x.isnull && y.isnull
-        return true
-    elseif x.isnull || y.isnull
-        return false
+
+## Operators
+
+"""
+    null_safe_op(f::Any, ::Type)::Bool
```

Comment:
`undef_op_is_safe`?

Don't spend too much time on this; if this feature will be useful in other places as well (`map` on an array of nullables?), then we can change the name later.